R/luz.R

```diff
@@ -0,0 +1,64 @@
+#' @rdname vetiver_create_description
+#' @export
+vetiver_create_description.luz_module_fitted <- function(model) {
+    glue("A luz module")
+}
+
+#' @rdname vetiver_create_meta
+#' @export
+vetiver_create_meta.luz_module_fitted <- function(model, metadata) {
+    pkgs <- c("luz", model$model$required_pkgs)
```

Comment:
Can you tell me more about how this works in luz? I have trained some of the example models and they do require me to have, for example, torch and/or torchvision loaded, but then they are not stored in this slot. Instead, I see:

```r
model$model$required_pkgs
#> NULL
```

Reply:
In luz we don't try to be smart about capturing used packages, but users can optionally set this field in the `nn_module` so it's available as metadata. Eg, one can do:

```
module <- torch::nn_module(
    initialize = function(in_features, out_features) {
        self$linear <- torch::nn_linear(10, 10)
    },
    forward = function(x) {
        self$linear(x)
    },
    required_pkgs = c("torch", "torchvision")
)
```

We could try to traverse the `forward` expression and find functions calls that come from other packages, but I feel this can still have many edge cases and is kind of error prone.

Reply:
Would they always need torch? Should we include that there? I think this is about what needs to be installed and attached for predictions to work. Getting the right packages installed for the deployment is a big part of what vetiver aims to do.